brouter-core/src/main/java/btools/router/RoutingEngine.java

```diff
@@ -447,6 +478,79 @@ public void doGetInfo() {
     }
   }
 
+  public void doRoundTrip() {
+    try {
+      startTime = System.currentTimeMillis();
+      MatchedWaypoint wpt1 = new MatchedWaypoint();
+      wpt1.waypoint = waypoints.get(0);
+      wpt1.name = "roundtrip";
+
+      routingContext.useDynamicDistance = true;
+      double searchRadius = (routingContext.roundtripDistance == null ? 1500 :routingContext.roundtripDistance);
+      double direction = (routingContext.startDirection == null ? -1 :routingContext.startDirection);
+      if (direction == -1) direction = (int) (Math.random()*360);
+      double directionAdd = (routingContext.roundtripDirectionAdd == null ? ROUNDTRIP_DEFAULT_DIRECTIONADD :routingContext.roundtripDirectionAdd);
+
+      if (routingContext.allowSamewayback) {
+        int[] pos = CheapRuler.destination(waypoints.get(0).ilon, waypoints.get(0).ilat, searchRadius, direction);
+        MatchedWaypoint wpt2 = new MatchedWaypoint();
+        wpt2.waypoint = new OsmNode(pos[0], pos[1]);
+        wpt2.name = "rt1_" + direction;
+
+        OsmNodeNamed onn = new OsmNodeNamed(new OsmNode(pos[0], pos[1]));
+        onn.name = "via1";
+        onn.radius = 1000;
+        waypoints.add(onn);
+      } else {
+        int[] pos = CheapRuler.destination(waypoints.get(0).ilon, waypoints.get(0).ilat, searchRadius, direction-directionAdd);
+        MatchedWaypoint wpt2 = new MatchedWaypoint();
+        wpt2.waypoint = new OsmNode(pos[0], pos[1]);
+        wpt2.name = "via1_" + (int) (direction-directionAdd);
+
+        OsmNodeNamed onn = new OsmNodeNamed(new OsmNode(pos[0], pos[1]));
+        onn.name = "via1";
+        onn.radius = 1000;
```

Comment:
@afischerdev Here why do we set a `radius`?
As this a via point, not a no-go.

Reply:
@devemux86 
The radius is also used for distance to the nearest road calculation.
I think it is from the beginning of this project, where the dynamic distance wasn't there and I tried to have a slightly larger radius.
But this is redundant now. I will remove it